api/v1/candidates.coffee

```diff
@@ -1 +1,47 @@
+ObjectID = require('mongodb').ObjectID
+
 module.exports = (api, db) ->
+  api = api || {}
+  api.v1 = api.v1 || {}
+
+  api.v1.getCandidate = (req, res) ->
+    db.collection("candidates").findOne( {_id: new ObjectID(req.params.id)}, (err, results) ->
+      if err
+        res.jsonp
+          error: err
+        , 500
+
+        return
+
+      res.jsonp(results)
+    )
+
+  api.v1.getCandidates = (req, res) ->
+    api.v1.search(req, res)
+
+  api.v1.search = (req, res) ->
+    findCriteria = req.query || {}
+    if findCriteria.nameId
+      findCriteria.nameId = parseInt(findCriteria.nameId) || findCriteria.nameId
+    db.collection("candidates").find(findCriteria).toArray (err, results) ->
```

Comment:
I hope I have understood this correctly. I am thinking the search parameters would be query string parameters in the URL. Is that correct?
If that assumption is correct, since our search criteria is not too complex, there might not be a need to dynamically build the query like that. Express does the magic of constructing a JSON out of the query string parameters. So for a URL like ```http://localhost:3000/api/v1/search?firstName=First%20Name%201&lastName=Last%20Name%201```, ```req.query``` would be ```{ firstName: 'First Name 1', lastName: 'Last Name 1' }```. For multiple clauses in the query JSON, MongoDB implicitly adds a logical ```AND``` (from the documentation at http://docs.mongodb.org/manual/tutorial/query-documents/). Does that handle the requirements for this API?
I tested this out with a few combinations of nameId, firstName and lastName. But do let me know if I missed something.